nixos/modules/programs/maven.nix

```diff
@@ -0,0 +1,30 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.maven;
+in {
+  options = {
+    programs.maven = {
+      enable = mkEnableOption "maven";
+
+      package = mkOption {
+        type = types.path;
+        description = "The maven package version";
+        default = pkgs.maven;
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment = {
+      systemPackages = [ cfg.package ];
+
+      variables = rec {
+        M2 = "${M2_HOME}/bin";
+        M2_HOME = "${cfg.package}/maven";
```

Comment:
Shouldn't M2_HOME be writable ?
This is where the .m2/repository gets written.

Reply:
`M2_HOME` is like `JAVA_HOME` it points to the root of the package.

<https://askubuntu.com/a/1052205>

Setting `M2_HOME` is not necessary but certain tools still use it for example IntelliJ.